examples/06_deduplication.py

```diff
@@ -0,0 +1,186 @@
+unique_examples, counts = np.unique(data, return_counts=True)
+# and build a series out of them
+ex_series = pd.Series(counts, index=unique_examples)
+
+###############################################################################
+# Visualize the data
+# ------------------
+
+ex_series.plot.barh(figsize=(10, 15))
+_ = plt.xlabel("Medication name")
+_ = plt.ylabel("Counts")
+
+###############################################################################
+# We can now see clearly the structure of the data: The three original medications
+# are the most common ones, however there are many spelling mistakes and hence
+# many slight variations of the names of the original medications.
+#
+# The idea is to use the fact that the string-distance of each misspelled medication
+# name will be closest to either the correctly or incorrectly spelled orginal
+# medication name - and therefore form clusters.
+
+
+from dirty_cat import deduplicate
+from dirty_cat._deduplicate import _compute_ngram_distance
+from scipy.spatial.distance import squareform
+
+ngram_distances = _compute_ngram_distance(unique_examples)
+square_distances = squareform(ngram_distances)
+
+###############################################################################
+# We can visualize the pair-wise distance between all medication names
+# --------------------------------------------------------------------
+#
+# Below I use a heatmap to visualize the pairwise-distance between medication names.
+# A darker color means that two medication names are closer together (i.e. more similar),
+# a lighter color means a larger distance. We can see that we are dealing with three
+# clusters - the original medication names and their misspellings that cluster around them.
+
+import seaborn as sns
+
+fig, axes = plt.subplots(1, 1, figsize=(12, 12))
+sns.heatmap(
+    square_distances, yticklabels=ex_series.index, xticklabels=ex_series.index, ax=axes
+)
+
+###############################################################################
+# Clustering to suggest corrections of misspelled names
+# -----------------------------------------------------
+#
+# The number of clusters will need some adjustment depending on the data you have.
+# If no fixed number of clusters is given, `deduplicate` tries to set it automatically
+# via the `silhouette score <https://scikit-learn.org/stable/modules/clustering.html#silhouette-coefficient>`_.
+
+
+deduplicated_data = deduplicate(data)
+
+###############################################################################
+# We can visualize the distribution of categories in the deduplicated data:
+
+deduplicated_unique_examples, deduplicated_counts = np.unique(
+    deduplicated_data, return_counts=True
+)
+deduplicated_series = pd.Series(deduplicated_counts, index=deduplicated_unique_examples)
+
+deduplicated_series.plot.barh(figsize=(10, 15))
+_ = plt.xlabel("Medication name")
+_ = plt.ylabel("Counts")
```

Comment:
Is there a specific reason why we need to get/store the function output? Otherwise it's much cleaner to do
```suggestion
plt.xlabel("Medication name")
plt.ylabel("Counts")
```

Reply:
Just personal preference to not print the label to stdout (and have it show up in the cell).
But it's probably easier to get rid of this idiosyncrasy :)